packages/gatsby/src/utils/worker/types.ts

```diff
@@ -0,0 +1,24 @@
+import type Worker from "jest-worker"
+
+type WrapReturnOfAFunctionInAPromise<
+  FunctionThatDoesNotReturnAPromise extends (...args: Array<any>) => any
+> = (
+  ...a: Parameters<FunctionThatDoesNotReturnAPromise>
+) => Promise<ReturnType<FunctionThatDoesNotReturnAPromise>>
+
+// jest-worker will make sync function async, so to keep proper types we need to adjust types so all functions
+// on worker pool are async
+type EnsureFunctionReturnsAPromise<MaybeFunction> = MaybeFunction extends (
+  ...args: Array<any>
+) => Promise<any>
+  ? MaybeFunction
+  : MaybeFunction extends (...args: Array<any>) => any
+  ? WrapReturnOfAFunctionInAPromise<MaybeFunction>
+  : never
+
+export type CreateWorkerPoolType<ExposedFunctions> = Worker &
+  {
+    [FunctionName in keyof ExposedFunctions]: EnsureFunctionReturnsAPromise<
+      ExposedFunctions[FunctionName]
+    >
+  }
```

Comment:
WorkerPool type creator, now used just for main workerPool, but in following PRs will be used for tests where test workerPool will include additional helper functions for tests